ArduCopter/mode_autotune.cpp

```diff
@@ -106,42 +104,44 @@ bool Copter::ModeAutoTune::init(bool ignore_checks)
 
         case UNINITIALISED:
             // autotune has never been run
-            success = start(false);
-            if (success) {
-                // so store current gains as original gains
-                backup_gains_and_initialise();
-                // advance mode to tuning
-                mode = TUNING;
-                // send message to ground station that we've started tuning
-                update_gcs(AUTOTUNE_MESSAGE_STARTED);
-            }
+            // so store current gains as original gains
+            backup_gains_and_initialise();
+            // advance mode to tuning
+            mode = TUNING;
+            // send message to ground station that we've started tuning
+            update_gcs(AUTOTUNE_MESSAGE_STARTED);
             break;
 
         case TUNING:
             // we are restarting tuning after the user must have switched ch7/ch8 off so we restart tuning where we left off
-            success = start(false);
-            if (success) {
-                // reset gains to tuning-start gains (i.e. low I term)
-                load_intra_test_gains();
-                // write dataflash log even and send message to ground station
-                Log_Write_Event(DATA_AUTOTUNE_RESTART);
-                update_gcs(AUTOTUNE_MESSAGE_STARTED);
-            }
+            // reset gains to tuning-start gains (i.e. low I term)
+            load_intra_test_gains();
+            // write dataflash log even and send message to ground station
+            Log_Write_Event(DATA_AUTOTUNE_RESTART);
+            update_gcs(AUTOTUNE_MESSAGE_STARTED);
             break;
 
         case SUCCESS:
             // we have completed a tune and the pilot wishes to test the new gains in the current flight mode
             // so simply apply tuning gains (i.e. do not change flight mode)
             load_tuned_gains();
             Log_Write_Event(DATA_AUTOTUNE_PILOT_TESTING);
-            break;
+            return;
     }
 
     // only do position hold if starting autotune from LOITER or POSHOLD
     use_poshold = (_copter.control_mode == LOITER || _copter.control_mode == POSHOLD);
     have_position = false;
 
-    return success;
+    // initialize vertical speeds and leash lengths
```

Comment:
I'm not sure if this has any impact, but before this was run prior to `backup_gains_and_initialise`/`load_intra_test_gains`.

Reply:
I don't believe there's an impact.

Reply:
Besides the previously identified order change, this has a new behavior in case `mode == SUCCESS`: the code below looks harmless to me, but wasn't run before, so either @rmackay9 or @lthall should look at it before merging.

Reply:
I think this is ok.